modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFieldLocalServiceTest.java

```diff
@@ -922,6 +923,110 @@ public void testAddOrUpdateCustomObjectField() throws Exception {
 			_objectDefinitionLocalService, new String[] {"C_A", "C_AA"});
 	}
 
+	@Test
+	public void testAddOrUpdateSystemObjectField() throws Exception {
+		ObjectDefinition modifiableSystemObjectDefinition =
+			ObjectDefinitionTestUtil.addModifiableSystemObjectDefinition(
+				TestPropsValues.getUserId(), null,
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				"Test", null, null,
+				LocalizedMapUtil.getLocalizedMap(RandomTestUtil.randomString()),
+				ObjectDefinitionConstants.SCOPE_SITE, null, 1,
+				_objectDefinitionLocalService, Collections.emptyList());
+
+		ObjectField systemObjectField =
+			_objectFieldLocalService.addOrUpdateSystemObjectField(
+				null, TestPropsValues.getUserId(), 0L,
+				modifiableSystemObjectDefinition.getObjectDefinitionId(),
+				ObjectFieldConstants.BUSINESS_TYPE_TEXT, null, null,
+				ObjectFieldConstants.DB_TYPE_STRING, false, false, "",
+				LocalizedMapUtil.getLocalizedMap("Able"), "able",
+				ObjectFieldConstants.READ_ONLY_FALSE, null, false, false,
+				Collections.emptyList());
+
+		Assert.assertEquals("able", systemObjectField.getDBColumnName());
+		Assert.assertEquals(
+			ObjectFieldConstants.DB_TYPE_STRING, systemObjectField.getDBType());
+		Assert.assertFalse(systemObjectField.isIndexed());
+		Assert.assertFalse(systemObjectField.isIndexedAsKeyword());
+		Assert.assertEquals(
+			StringPool.BLANK, systemObjectField.getIndexedLanguageId());
+		Assert.assertEquals(
+			LocalizedMapUtil.getLocalizedMap("Able"),
+			systemObjectField.getLabelMap());
+		Assert.assertEquals("able", systemObjectField.getName());
+		Assert.assertFalse(systemObjectField.isRequired());
+		Assert.assertTrue(systemObjectField.isSystem());
```

Comment:
@seltong can we create a new private method `_assertSystemObjectField` with all these asserts to avoid duplicate code?

Reply:
Done